mitodl/mitxpro
# feat: add discount_type field and dollars-off support in CouponPaymentVersion and APIs

#### Pre-Flight checklist


#### What are the relevant tickets?
#2446 

#### What's this PR do?
- Adds support for discount type for dollars-off
- Adds new field in `discount_type` CouponPaymentVersion
- refactors existing tests to support dollars-off
- Adds amount field validators based on discount type
- Refactors serializers and Django admin support for dollars-off coupon creation

#### How should this be manually tested?
- Before shifting to this PR, Make sure you have created some coupons e.g. Global, Non-Global, with some amount. 
- Moving to this PR and after `docker-compose up` you should see the migrations running successfully
- Go to CouponPaymentVersion in django admin and check that there is a new dropdown with `percent-off` and `dollars-off`
- Check the `discount_type` in your previous CouponPaymentVersions are set `percent-off` by default
- Try to create `dollars-off` coupon with 0 amount and you should see a proper message in Admin
- Try to create `percent-off` coupon with a value of more than 1 and you should see a proper error message
- Now try to create valid coupons for both `percent-off` and `dollars-off` and they should be created successfully
- Try the `api/coupons/` API with POST call and make sure that you see validation errors for the amount like mentioned above points
- 

#### Where should the reviewer start?
- Create and test a few coupons before moving to this PR
